traversal/planner/PlannerEdge.java

```diff
@@ -186,6 +194,14 @@ void createOptimiserConstraints() {
             conIsSelected.setCoefficient(to.varOrderNumber, -1);
 
             int numAdjacent = to.ins().size();
+
+            for (int i = 0; i < numAdjacent; i++) {
+                OptimiserConstraint conMinimalWithMultiplicity = planner.optimiser().constraint(0, 1, conPrefix + "minimal_with_multiplicity[" + i + "]");
+                conMinimalWithMultiplicity.setCoefficient(varIsMinimal, 1);
+                conMinimalWithMultiplicity.setCoefficient(to.varNumInsEncoding[i + 1], 1);
+                conMinimalWithMultiplicity.setCoefficient(varIsMinimalWithMultiplicity[i], -2);
+            }
```

Comment:
Edge is minimal-with-multiplicity-i iff 1) it is minimal, and 2) its destination vertex has i+1 incoming edges.